library/Core/src/Juvix/Core/IR/CheckTerm.hs

```diff
@@ -179,7 +180,11 @@ typeTerm' term ann@(Typed.Annotation σ ty) =
     Core.PrimTy t _ -> do
       requireZero σ
       void $ requirePrimStars (Param.primArity t) ty
-      let t' = App.Return {retTerm = t, retType = Param.PrimType $ Param.STAR :| []}
+      let t' =
+            App.Return
+              { retTerm = t,
+                retType = Param.PrimType $ NonEmpty.fromList $ replicate (fromIntegral (Param.primArity t + 1)) Param.STAR
```

Comment:
A recursive function of over `ty` should also work, I guess. Would that be nicer?